vantage/access_grant_resource.go

```diff
@@ -0,0 +1,175 @@
+package vantage
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	modelsv2 "github.com/vantage-sh/vantage-go/vantagev2/models"
+	accessgrantsv2 "github.com/vantage-sh/vantage-go/vantagev2/vantage/access_grants"
+)
+
+type AccessGrantResource struct {
+	client *Client
+}
+
+func NewAccessGrantResource() resource.Resource {
+	return &AccessGrantResource{}
+}
+
+type AccessGrantResourceModel struct {
+	Token         types.String `tfsdk:"token"`
+	ResourceToken types.String `tfsdk:"resource_token"`
+	TeamToken     types.String `tfsdk:"team_token"`
+	Access        types.String `tfsdk:"access"`
+}
+
+func (r *AccessGrantResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_access_grant"
+}
+
+func (r AccessGrantResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"resource_token": schema.StringAttribute{
+				MarkdownDescription: "Token of the resource being granted.",
+				Required:            true,
+			},
+			"team_token": schema.StringAttribute{
+				MarkdownDescription: "Token of the team being granted.",
+				Required:            true,
+			},
+			"access": schema.StringAttribute{
+				// TODO(ac): Implement CustomType that validates "allowed" or "denied".
```

Comment:
Would there need to be a terraform state migration if this is switched from `string` to a custom type? If so, probably worth doing now to avoid that in the future otherwise it'll likely never happen :)

Reply:
turns out there was an easier way: https://github.com/vantage-sh/terraform-provider-vantage/pull/11/commits/b70c74d687fa2841a478cfdcd41eb65f11b58a1a